deploy/ControllerInit.sol

```diff
@@ -67,18 +71,45 @@ library MainnetControllerInit {
+
+        require(almProxy.hasRole(DEFAULT_ADMIN_ROLE, params.admin)   == true, "MainnetControllerInit/incorrect-admin-almProxy");
+        require(controller.hasRole(DEFAULT_ADMIN_ROLE, params.admin) == true, "MainnetControllerInit/incorrect-admin-controller");
+        require(rateLimits.hasRole(DEFAULT_ADMIN_ROLE, params.admin) == true, "MainnetControllerInit/incorrect-admin-rateLimits");
+
+        require(address(controller.proxy())      == controllerInst.almProxy,   "MainnetControllerInit/incorrect-almProxy");
+        require(address(controller.rateLimits()) == controllerInst.rateLimits, "MainnetControllerInit/incorrect-rateLimits");
+        require(address(controller.vault())      == ilkInst.vault,             "MainnetControllerInit/incorrect-vault");
+        require(address(controller.buffer())     == ilkInst.buffer,            "MainnetControllerInit/incorrect-buffer");
+        require(address(controller.psm())        == params.psm,                "MainnetControllerInit/incorrect-psm");
+        require(address(controller.daiUsds())    == params.daiUsds,            "MainnetControllerInit/incorrect-daiUsds");
+        require(address(controller.cctp())       == params.cctpMessenger,      "MainnetControllerInit/incorrect-cctpMessenger");
+        require(address(controller.susds())      == params.susds,              "MainnetControllerInit/incorrect-susds");
+        require(address(controller.dai())        == params.dai,                "MainnetControllerInit/incorrect-dai");
+        require(address(controller.usdc())       == params.usdc,               "MainnetControllerInit/incorrect-usdc");
+        require(address(controller.usds())       == params.usds,               "MainnetControllerInit/incorrect-usds");
 
-        controller.grantRole(controller.FREEZER(), freezer);
-        controller.grantRole(controller.RELAYER(), relayer);
+        require(controller.psmTo18ConversionFactor() == 1e12, "MainnetControllerInit/incorrect-psmTo18ConversionFactor");
+        require(controller.active()                  == true, "MainnetControllerInit/controller-not-active");
+
+        // Step 2: Configure ACL permissions for controller and almProxy
+
+        controller.grantRole(controller.FREEZER(), params.freezer);
+        controller.grantRole(controller.RELAYER(), params.relayer);
 
         almProxy.grantRole(almProxy.CONTROLLER(), address(controller));
 
         rateLimits.grantRole(rateLimits.CONTROLLER(), address(controller));
 
-        // Step 2: Configure all rate limits for controller, using Base as only domain
+        // Step 3: Configure all rate limits for controller, using Base as only domain
+
+        // Sanity check all rate limit data
+        _checkUnlimitedData(usdsMintData,         "usdsMintData");
+        _checkUnlimitedData(usdcToUsdsData,       "usdcToUsdsData");
+        _checkUnlimitedData(usdcToCctpData,       "usdcToCctpData");
+        _checkUnlimitedData(cctpToBaseDomainData, "cctpToBaseDomainData");
```

Comment:
I think it will be easier to just wrap setting the parameter altogether.

`_setRateLimitData(bytes32 key, RateLimitData memory data)` and probably even an overloaded one for the domain id version. Inside that function should check unlimited is set properly.

Reply:
Updated